x-pack/platform/plugins/shared/triggers_actions_ui/public/application/sections/action_connector_form/connector_form.tsx

```diff
@@ -76,17 +77,17 @@ const formDeserializer = (data: ConnectorFormSchema): ConnectorFormSchema => {
     return data;
   }
 
-  const webhookData = data as { config: { headers?: Record<string, string> } };
-  const headers = Object.entries(webhookData?.config?.headers ?? {}).map(([key, value]) => ({
+  const configHeaders = Object.entries(data?.config.headers ?? {}).map(([key, value]) => ({
     key,
     value,
+    type: 'config' as const,
   }));
 
   return {
-    ...data,
-    config: {
-      ...data.config,
-      headers: isEmpty(headers) ? undefined : headers,
+    ...(data as any),
```

Comment:
I think instead of casting better, to change the types of the `formSerializer` to reflect the actual structure of the form. Same for below.